### ISSUE TITLE
Debian 11: Internet Connection Completely Blocked After Installing Mullvad 2022.5

### ISSUE THREAD

**Issue by cpunk** [mullvad/mullvadvpn-app#4190]
## Issue report

Operating system: Debian 11 (Bullseye)

App version: 2022.5

### Issue description

After installing Mullvad 2022.5 the complete internet access is blocked. When starting the app, not even the account name is requested, because the Mullvad servers are no longer accessible. Previously v2022.4 was installed and everything worked.

The installation process looks like this:
```
Removed /etc/systemd/system/multi-user.target.wants/mullvad-daemon.service.
Failed to disable unit: Unit file mullvad-early-boot-blocking.service does not exist.
Entpacken von mullvad-vpn (2022.5) über (2022.4) ...
mullvad-vpn (2022.5) wird eingerichtet ...
Created symlink /etc/systemd/system/multi-user.target.wants/mullvad-daemon.service → /usr/lib/systemd/system/mullvad-daemon.service.
Created symlink /etc/systemd/system/mullvad-daemon.service.wants/mullvad-early-boot-blocking.service → /usr/lib/systemd/system/mullvad-early-boot-blocking.service.
```

Reinstalled 2022.4 and erverything works fine again.

**Comment by pinkisemils** [comment]
Would you mind sending a problem report?

**Comment by pinkisemils** [comment]
I cannot reproduce this on Debian 11.

**Comment by cpunk** [comment]
Before I installed the version 2022.5, the daemon was stopped via systemctl stop mullvad-daemon.service.

Then I installed 2022.5 via sudo apt install -y ./MullvadVPN-2022.5_amd64.deb. After that the complete internet access of the computer is blocked. 

/etc/syslog: 
Feb  2 16:56:05 xy mullvad-daemon[4527]: [2023-02-02 16:56:05.498][talpid_core::firewall][#033[32mINFO#033[0m] Applying firewall policy: Blocked. Allowing LAN. Allowing endpoint: none
Feb  2 16:56:05 xy systemd[1]: mullvad-early-boot-blocking.service: Succeeded.
Feb  2 16:56:05 xy systemd[1]: Finished Mullvad early boot network blocker.
Feb  2 16:56:05 xy systemd[4539]: mullvad-daemon.service: Failed to locate executable /opt/Mullvad VPN/resources/mullvad-daemon: No such file or directory
Feb  2 16:56:05 xy systemd[4539]: mullvad-daemon.service: Failed at step EXEC spawning /opt/Mullvad VPN/resources/mullvad-daemon: No such file or directory
Feb  2 16:56:05 xy systemd[1]: Started Mullvad VPN daemon.
Feb  2 16:56:05 xy systemd[1]: mullvad-daemon.service: Main process exited, code=exited, status=203/EXEC
Feb  2 16:56:05 xy systemd[1]: mullvad-daemon.service: Failed with result 'exit-code'.
Feb  2 16:56:06 xy systemd[1]: mullvad-daemon.service: Scheduled restart job, restart counter is at 4.
Feb  2 16:56:06 xy systemd[1]: Stopped Mullvad VPN daemon.
Feb  2 16:56:06 xy systemd[1]: Starting Mullvad early boot network blocker...
Feb  2 16:56:06 xy mullvad-daemon[4540]: [2023-02-02 16:56:06.745][mullvad_daemon::version][#033[32mINFO#033[0m] Starting mullvad-daemon - 2022.5 2022-10-14
Feb  2 16:56:06 xy mullvad-daemon[4540]: [2023-02-02 16:56:06.746][mullvad_daemon::settings][#033[32mINFO#033[0m] Loading settings from /etc/mullvad-vpn/settings.json
Feb  2 16:56:06 xy mullvad-daemon[4540]: [2023-02-02 16:56:06.746][mullvad_daemon::early_boot_firewall][#033[32mINFO#033[0m] Applying firewall policy Blocked. Allowing LAN. Allowing endpoint: none
Feb  2 16:56:06 xy mullvad-daemon[4540]: [2023-02-02 16:56:06.747][talpid_core::firewall][#033[32mINFO#033[0m] Applying firewall policy: Blocked. Allowing LAN. Allowing endpoint: none
Feb  2 16:56:06 xy systemd[1]: mullvad-early-boot-blocking.service: Succeeded.
Feb  2 16:56:06 xy systemd[1]: Finished Mullvad early boot network blocker.
Feb  2 16:56:06 xy systemd[1]: Started Mullvad VPN daemon.
Feb  2 16:56:06 xy systemd[4552]: mullvad-daemon.service: Failed to locate executable /opt/Mullvad VPN/resources/mullvad-daemon: No such file or directory
Feb  2 16:56:06 xy systemd[4552]: mullvad-daemon.service: Failed at step EXEC spawning /opt/Mullvad VPN/resources/mullvad-daemon: No such file or directory
Feb  2 16:56:06 xy systemd[1]: mullvad-daemon.service: Main process exited, code=exited, status=203/EXEC
Feb  2 16:56:06 xy systemd[1]: mullvad-daemon.service: Failed with result 'exit-code'.
Feb  2 16:56:07 xy systemd[1]: mullvad-daemon.service: Scheduled restart job, restart counter is at 5.
Feb  2 16:56:07 xy systemd[1]: Stopped Mullvad VPN daemon.
Feb  2 16:56:07 xy systemd[1]: Starting Mullvad early boot network blocker...
Feb  2 16:56:07 xy mullvad-daemon[4553]: [2023-02-02 16:56:07.997][mullvad_daemon::version][#033[32mINFO#033[0m] Starting mullvad-daemon - 2022.5 2022-10-14
Feb  2 16:56:07 xy mullvad-daemon[4553]: [2023-02-02 16:56:07.998][mullvad_daemon::settings][#033[32mINFO#033[0m] Loading settings from /etc/mullvad-vpn/settings.json
Feb  2 16:56:07 xy mullvad-daemon[4553]: [2023-02-02 16:56:07.998][mullvad_daemon::early_boot_firewall][#033[32mINFO#033[0m] Applying firewall policy Blocked. Allowing LAN. Allowing endpoint: none
Feb  2 16:56:07 xy mullvad-daemon[4553]: [2023-02-02 16:56:07.998][talpid_core::firewall][#033[32mINFO#033[0m] Applying firewall policy: Blocked. Allowing LAN. Allowing endpoint: none
Feb  2 16:56:08 xy systemd[1]: mullvad-early-boot-blocking.service: Succeeded.
Feb  2 16:56:08 xy systemd[1]: Finished Mullvad early boot network blocker.
Feb  2 16:56:08 xy systemd[4565]: mullvad-daemon.service: Failed to locate executable /opt/Mullvad VPN/resources/mullvad-daemon: No such file or directory
Feb  2 16:56:08 xy systemd[4565]: mullvad-daemon.service: Failed at step EXEC spawning /opt/Mullvad VPN/resources/mullvad-daemon: No such file or directory
Feb  2 16:56:08 xy systemd[1]: Started Mullvad VPN daemon.
Feb  2 16:56:08 xy systemd[1]: mullvad-daemon.service: Main process exited, code=exited, status=203/EXEC
Feb  2 16:56:08 xy systemd[1]: mullvad-daemon.service: Failed with result 'exit-code'.
Feb  2 16:56:09 xy systemd[1]: mullvad-daemon.service: Scheduled restart job, restart counter is at 6.
Feb  2 16:56:09 xy systemd[1]: Stopped Mullvad VPN daemon.
Feb  2 16:56:09 xy systemd[1]: mullvad-early-boot-blocking.service: Start request repeated too quickly.
Feb  2 16:56:09 xy systemd[1]: mullvad-early-boot-blocking.service: Failed with result 'start-limit-hit'.
Feb  2 16:56:09 xy systemd[1]: Failed to start Mullvad early boot network blocker.
Feb  2 16:56:09 xy systemd[1]: mullvad-daemon.service: Start request repeated too quickly.
Feb  2 16:56:09 xy systemd[1]: mullvad-daemon.service: Failed with result 'exit-code'.
Feb  2 16:56:09 xy systemd[1]: Failed to start Mullvad VPN daemon.

It looks like the daemon expected in /opt/ Mullvad VPN/ is not found.

**Comment by faern** [comment]
>  Feb 2 16:56:06 xy systemd[4552]: mullvad-daemon.service: Failed at step EXEC spawning /opt/Mullvad VPN/resources/mullvad-daemon: No such file or directory

`mullvad-daemon` was moved and is now installed under `/usr/bin/mullvad-daemon`. For some reason your systemd unit file is not updated. Are you getting this from a clean install of the app?

**Comment by cpunk** [comment]
> Are you getting this from a clean install of the app?

First I installed it over the version 2022.4. When that didn't work, I uninstalled the current version again and went back to 2022.4. I think I had also completely uninstalled both versions in one of my attempts, but will try it again and report back here. Otherwise, I'll try to delete the systemd script by hand, or adjust it myself. Thanks for the tip!

**Comment by cpunk** [comment]
Now it works, thanks again for the hint, @faern 
I had not considered that the uninstall script does not delete all necessary files: 

`sudo apt purge mullvad-vpn`
Removed /etc/systemd/system/multi-user.target.wants/mullvad-daemon.service.
Removing mullvad-vpn app settings from /home/xy/.config/Mullvad VPN

This command did **not** remove the file /etc/systemd/system/mullvad-daemon.service

I have deleted it by hand and the installation of version 2022.5 has now worked. Possibly this is a topic for another bugreport?
